\documentclass{beamer}
\setbeameroption{hide notes}

%\documentclass[handout]{beamer}
%\setbeameroption{show notes}

%\documentclass{beamer}
%\usepackage{pgfpages}
%\setbeameroption{show notes on second screen}

%\documentclass[a4paper]{article}
%\usepackage{beamerarticle}

% Based on beamer solution template "generic-ornate-15min-45min.en.tex" by Till Tantau.

\mode<presentation>
{
  \usetheme{Berkeley}
%  \usetheme{Berlin}
%  \usetheme{Frankfurt}
%  \usetheme{Warsaw}
%  \usetheme{Ilmenau}

%  \usecolortheme{seahorse}
%  \usecolortheme{dolphin}
%  \usecolortheme{whale}

%  \usecolortheme{rose}
%  \usecolortheme{orchid}

  \usefonttheme{default}

  \setbeamercovered{transparent}

}


\usepackage[english]{babel}

\usepackage[latin1]{inputenc}

%\usepackage{times}
%\usepackage[T1]{fontenc}
\usepackage{lmodern}
\usepackage[T1]{fontenc}



% support for \includesvg
\newcommand{\executeiffilenewer}[3]{%
\ifnum\pdfstrcmp{\pdffilemoddate{#1}}%
{\pdffilemoddate{#2}}>0%
{\immediate\write18{#3}}\fi%
}
\newcommand{\includesvg}[1]{%
\executeiffilenewer{#1.svg}{#1.pdf}%
{inkscape -z -D --file=#1.svg %
--export-pdf=#1.pdf --export-dpi=72 --export-latex}%
\input{#1.pdf_tex}%
}



%\usepackage{caption}
%\usepackage{listings}
% setup listings captions
%\DeclareCaptionFont{white}{ \color{white} }
%\DeclareCaptionFormat{listing}{
%  \colorbox[cmyk]{0.43, 0.35, 0.35,0.01 }{
%    \parbox{\textwidth}{\hspace{15pt}#1#2#3}
%  }
%}
%\captionsetup[lstlisting]{ format=listing, labelfont=white, textfont=white, singlelinecheck=false, margin=0pt, font={bf,footnotesize} }

\usepackage[listings,theorems]{tcolorbox}

% custom listing styles
\lstdefinestyle{custombash}{
  belowcaptionskip=1\baselineskip,
  breaklines=true,
%  frame=L,
%  xleftmargin=\parindent,
  language=sh,
  showstringspaces=true,
  basicstyle=\tiny\ttfamily,
  keywordstyle=\bfseries\color{green!40!black},
  commentstyle=\itshape\color{purple!40!black},
  identifierstyle=\color{blue},
  stringstyle=\color{orange},
  escapeinside={+-}{-+}
}

\lstdefinestyle{customsam}{
  belowcaptionskip=1\baselineskip,
  breaklines=true,
%  frame=L,
%  xleftmargin=\parindent,
%  language=SAM,
  showstringspaces=true,
  basicstyle=\tiny\ttfamily,
  keywordstyle=\bfseries\color{green!40!black},
  commentstyle=\itshape\color{purple!40!black},
  identifierstyle=\color{blue},
  stringstyle=\color{orange},
}


\usepackage{path}

\newcommand{\coursedate} {11th October 2017}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Title, Author, etc
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\title[EBI NGS Workshop]{EBI Next Generation Sequencing Workshop}
\subtitle{Resequencing and variant calling}

\author[]%[J.~C.~Randall]
{Joshua~C.~Randall}

\institute[Wellcome Trust Sanger Institute]
{
  Senior Scientific Manager\\
  Human Genetics Informatics\\
  Wellcome Trust Sanger Institute
}

%\subject{NGS resequencing, assembly, and variant calling}
% This is only inserted into the PDF information catalog. Can be left out.

%\pgfdeclareimage[height=0.5cm]{sanger-logo}{Sang_Log_Lge_2col}
%\logo{\pgfuseimage{sanger-logo}}
\pgfdeclareimage[height=0.5cm, width=0.97cm]{hgi-logo}{HGI-ltblue-nostroke}
\logo{\pgfuseimage{hgi-logo}}


% Delete this, if you do not want the table of contents to pop up at
% the beginning of each subsection:
\AtBeginSubsection[]
{
  \begin{frame}<beamer>{Outline}
    \tableofcontents[currentsection,currentsubsection]
%    \tableofcontents[currentsection]
  \end{frame}
}


% If you wish to uncover everything in a step-wise fashion, uncomment
% the following command:

%\beamerdefaultoverlayspecification{<+->}


\begin{document}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Title
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}
  \titlepage
\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Outline
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Outline}
  \tableofcontents
  % You might wish to add the option [pausesections]
\end{frame}


% Since this a solution template for a generic talk, very little can
% be said about how it should be structured. However, the talk length
% of between 15min and 45min and the theme suggest that you stick to
% the following rules:
%
% - Exactly two or three sections (other than the summary).
% - At *most* three subsections per section.
% - Talk about 30s to 2min per frame. So there should be between about
%   15 and 30 frames, all told.

%\section{Introduction}
%
%\subsection[Short First Subsection Name]{First Subsection Name}
%
%\begin{frame}{Make Titles Informative. Use Uppercase Letters.}{Subtitles are optional.}
%  % - A title should summarize the slide in an understandable fashion
%  %   for anyone how does not follow everything on the slide itself.
%
%  \begin{itemize}
%  \item
%    Use \texttt{itemize} a lot.
%  \item
%    Use very short sentences or short phrases.
%  \end{itemize}
%\end{frame}
%
%\begin{frame}{Make Titles Informative.}
%
%  You can create overlays\dots
%  \begin{itemize}
%  \item using the \texttt{pause} command:
%    \begin{itemize}
%    \item
%      First item.
%      \pause
%    \item
%      Second item.
%    \end{itemize}
%  \item
%    using overlay specifications:
%    \begin{itemize}
%    \item<3->
%      First item.
%    \item<4->
%      Second item.
%    \end{itemize}
%  \item
%    using the general \texttt{uncover} command:
%    \begin{itemize}
%      \uncover<5->{\item
%        First item.}
%      \uncover<6->{\item
%        Second item.}
%    \end{itemize}
%  \end{itemize}
%\end{frame}
%
%
%\subsection{Second Subsection}
%
%\begin{frame}{Make Titles Informative.}
%\end{frame}
%
%\begin{frame}{Make Titles Informative.}
%\end{frame}
%\section{Lecture}


%\section{Practical}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Exercises: Mapping
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Exercises: Mapping}

\begin{frame}[fragile]{Exercises: Mapping}
%\begin{itemize}
%\item
%\end{itemize}
\begin{tcblisting}{title={Prepare Exercise Data}, listing only, listing options={style=custombash}}
# expand archive
tar+-\quad -+zxvf+-\quad -+EBI_NGS_rsvc_lab.tgz
# change into lab directory
cd+-\quad -+EBI_NGS_rsvc_lab
\end{tcblisting}
\footnotesize{Thanks to Thomas Keane (WTSI Vertebrate Resequencing) for providing the data and developing the initial version of these exercises. }
\end{frame}

\subsection{BWA Alignment}
\begin{frame}[fragile]{Exercises: Mapping}
\framesubtitle{BWA Alignment}
\begin{itemize}
\item \uncover<1->{Index the reference genome with bwa}
\only<1>{
         \begin{tcolorbox}[fontupper=\scriptsize]
         Before aligning to the reference, the reference needs to indexed. Change to the `\path !ref!' directory and use the `\texttt{bwa index}' command to index the reference genome fasta (\texttt{.fa}). \\
         Running `\texttt{bwa index}' with no options will give a help message -- what `\texttt{-a}' parameter should you use? (hint: yeast is considered a small genome).
         \end{tcolorbox}
}
\note<1>[item]{
	Index the reference genome with bwa:\\
	\texttt{cd\quad ref}\\
	\texttt{bwa\quad index\quad -a\quad is\quad S.cerevisiae.EF4.68.dna.toplevel.fa}
}

\item \uncover<2->{Align the lane fastq files with bwa}
\only<2>{
         \begin{tcolorbox}[fontupper=\scriptsize]
         Change to the \path !Exercise-align/60A-Sc-DBVPG6044/lane1! directory and use bwa to align the fastq files, generating \path !.sai! files for each. (hint: use `\texttt{bwa aln}' to align each fastq file).
         \end{tcolorbox}
}
\note<2>[item]{
	Align the lane fastq files with bwa:\\
	\texttt{bwa\quad aln\quad ../../../ref/S.cerevisiae.EF4.68.dna.toplevel.fa s-7-1.fastq\quad >\quad s-7-1.fastq.sai}\\
	\texttt{bwa\quad aln\quad ../../../ref/S.cerevisiae.EF4.68.dna.toplevel.fa s-7-2.fastq\quad >\quad s-7-2.fastq.sai}
}

\item \uncover<3->{Generate paired-end alignment with bwa}
\only<3>{
         \begin{tcolorbox}[fontupper=\scriptsize]
         Create a SAM file called \path !lane1.sam! with read group ID `\texttt{lane1}' referring to sample `\texttt{60A-Sc-DBVPG6044}' (hint: use `\texttt{bwa sampe}' to create the SAM file, using the option to set the read group header line to `\texttt{$@$RG$\backslash$tID:lane1$\backslash$tSM:60A-Sc-DBVPG6044}')
         \end{tcolorbox}
}
\note<3>[item]{
	Generate paired-end alignment with bwa:\\
	\texttt{bwa\quad sampe\quad -r\quad '$@$RG$\backslash$tID:lane1$\backslash$tSM:60A-Sc-DBVPG6044'\quad -a\quad 1000\quad ../../../ref/S.cerevisiae.EF4.68.dna.toplevel.fa\quad s-7-1.fastq.sai\quad s-7-2.fastq.sai\quad s-7-1.fastq\quad s-7-2.fastq\quad >\quad lane1.sam}
}

\item \uncover<4->{Convert to a BAM file}
\only<4>{
         \begin{tcolorbox}[fontupper=\scriptsize]
         The SAM file you created is human readable (try viewing it with `\texttt{less -S}') but is not a very efficient way for the computer to store the data. Use \texttt{samtools} to convert the SAM to a BAM (let's call it \path !lane1.bam!).
         (hint: perform the conversion using `\texttt{samtools view}')
         \end{tcolorbox}
}
\note<4>[item]{
	Convert to a BAM file:\\
	\texttt{samtools\quad view\quad -bS\quad \path !lane1.sam!\quad >\quad \path !lane1.bam!}
}

\item \uncover<5->{Sort and index the BAM file}
\only<5>{
         \begin{tcolorbox}[fontupper=\scriptsize]
         The BAM file you just created makes fairly efficient use of space, but can only be read by reading through the whole file. In order to be able to quickly find parts of the file that are of interest (such as regions in which to perform variant calling) we need to sort (by reference coordinate position) and produce an index of the sorted file for fast lookups of regions of interest. Use `\texttt{samtools sort}' to produce a new BAM file (call it \path !lane1.sorted.bam!) and then run `\texttt{samtools index}' on that file to create an index file alongside it.
         \end{tcolorbox}
}
\note<5>[item]{
	Sort and index the BAM file:\\
	\texttt{samtools\quad sort\quad \path !lane1.bam!\quad > \path !lane1.sorted!}\\
	\texttt{samtools\quad index\quad \path !lane1.sorted.bam!}
}

\end{itemize}
\end{frame}


\subsection{Generate QC Stats}
\begin{frame}[fragile]{Exercises}
\framesubtitle{Generate QC Stats}
\begin{itemize}
\item {Run samtools flagstat}
         \begin{tcolorbox}[fontupper=\scriptsize]
         We can use samtools to report basic statistics about the alignments in the BAM file we created in the last exercise. Run `\texttt{samtools flagstat}' on \path !lane1.sorted.bam!.
         \end{tcolorbox}
\item Looking at the flagstat output, answer the following questions:
	\begin{itemize}
	\item What is the total number of reads?
	\item What proportion of the reads were mapped?
	\item How many reads were paired correctly/properly?
	\item How many reads mapped to a different chromosome with map quality >=5?
	\end{itemize}
\end{itemize}
\end{frame}

 \subsection{QC Diagnostic Plots}
\begin{frame}[fragile]{Exercises}
\framesubtitle{QC Diagnostic Plots}
\begin{itemize}
\item Run samtools stats (previously bamcheck)
         \begin{tcolorbox}[fontupper=\scriptsize]
         Run \texttt{samtools stats} on \path !lane1.sorted.bam! and redirect \\
         output to \path !lane1.sorted.bam.bc!
         \end{tcolorbox}
\item Generate plots
         \begin{tcolorbox}[fontupper=\scriptsize]
         \texttt{plot-bamstats\quad -p\quad bamstats/\quad lane1.sorted.bam.bc}
         \end{tcolorbox}
\item View png plots (e.g. using `\texttt{xdg-open bamstats/*.png}')
\item Looking at the plots, answer the following questions:
	\begin{itemize}
	\item Does GC content vary between the fwd and rev reads?
	\item What is the median insert size?
	\item Which has higher average quality, fwd or rev reads?
	\item What can you tell about reverse read quality distribution?
	\end{itemize}
\end{itemize}
\end{frame}


\subsection{BWA Alignment Redux}
\begin{frame}[fragile]{Exercises}
\framesubtitle{BWA Alignment Redux}
\begin{itemize}
\item Two more fastq files from a second lane can be found in \path !Exercise-align/60A-Sc-DBVPG6044/lane2/!
\item Everything should be the same as last time around, except this time for the read group, use the ID `\texttt{lane2}' (it is still the same sample)
\item Now that you're an expert on samtools, can you manage to do a new paired-end alignment  without referring back to the previous section? (it's ok if you need to refer back, but try to do as much as you can on your own)
\item Your output file should be a sorted and indexed BAM file called `\path !lane2.sorted.bam!'
\end{itemize}
\end{frame}


\subsection{Merge BAMs}
\begin{frame}[fragile]{Exercises}
\framesubtitle{Merge BAMs}
\begin{itemize}
\item Change to the \path !Exercise-align/60A-Sc-DBVPG6044/! directory
\item Here there are two directories called \path !lane1! and \path !lane2!
\item Use the picard tool called `\texttt{MergeSamFiles}' to merge the two sorted lane BAM files into one library BAM called \path !library.bam!
         \begin{tcolorbox}[fontupper=\scriptsize]
         Find the options for `MergeSamFiles' by running: \\
         \texttt{picard MergeSamFiles}
         \end{tcolorbox}
\end{itemize}
\note<1->[item]{
	Merge BAMs:\\
	\texttt{picard\quad MergeSamFiles\quad I=lane1/lane1.sorted.bam\quad I=lane2/lane2.sorted.bam\quad O=library.bam}
}
\end{frame}


\subsection{Mark Duplicates}
\begin{frame}[fragile]{Exercises}
\framesubtitle{Mark Duplicates}
\begin{itemize}
\item Use the picard tool called `\texttt{MarkDuplicates}' to remove PCR duplicates that may have been introduced during the library construction stage (let's call the output file \path !library.markdup.bam!)
\item Don't forget to index the final output file using `\texttt{samtools index}'
\item How many reads were marked as duplicates?
\end{itemize}
\note<1->[item]{
	Mark Duplicates:\\
	\texttt{picard\quad MarkDuplicates\quad I=library.bam\quad O=library.markdup.bam\quad M=metrics.out}\\
	\texttt{10270 reads marked}\\
	\texttt{samtools\quad index\quad library.markdup.bam}
}
\end{frame}


\subsection{BAM Visualisation}
\begin{frame}[fragile]{Exercises}
\framesubtitle{BAM Visualisation}
IGV (\url{http://www.broadinstitute.org/igv/}) is a Java visualisation tool for looking at alignments of reads onto a reference genome from BAM files
\begin{itemize}
\item Launch IGV (using the Desktop shortcut)
\item Load the reference genome located in \path !ref! by going to the `\texttt{Genome}' menu and selecting `\texttt{Load Genome From File...}'
\item Load the \path !library.markdup.bam! BAM file by going to the `\texttt{File}' menu and selecting `\texttt{Load From File...}'
\item Go to chromosome IV and position $764,294$ using the top navigation bar
\item What is the reference base at this position?
\item Do the reads agree with the reference base?
\item How about at position $764,292$?
\end{itemize}
\end{frame}



\section{Exercises: Calling}
\subsection{Discovery with mpileup}
\begin{frame}[fragile]{Exercises: Calling}
\framesubtitle{Discovery with mpileup}
We will call SNPs and INDELs using `\texttt{samtools mpileup}' and `\texttt{bcftools}' -- let's first see how mpileup works
\begin{itemize}
\item Change to the \path !Exercise-varcall/! directory
\item Run `\texttt{samtools mpileup}' to see the available options
\item Run mpileup using the faidx reference sequence file \path !19.fa! and the input BAM file \path !A-J.19.10000000-11000000.bam! and pipe the output through \texttt{less}
\item Each line of mpileup output corresponds to a position on the genome and the columns are: chr, pos, reference base, read depth, read bases (`.' or `,' indicating reference match and `\$' indicating the end of a read) and base qualities encoded into characters.
\item What is the read depth at position 10000177?
\item What is the reference base?
\end{itemize}
\end{frame}


\subsection{Calling with bcftools}
\begin{frame}{Exercises: Calling}
\framesubtitle{Calling with bcftools}
\begin{itemize}
\item We're almost ready to call variants -- run `\texttt{bcftools call}' to get the options for bcftools variant calling (this used to be part of `\texttt{bcftools view}')
\item Now tell mpileup to generate BCF output and pipe the output to `\texttt{bcftools}' with options set to use the multiallelic caller to call both SNPs and INDELs (hint: don't skip indels and they will be called) and to only output potential variant sites (i.e. don't output sites where only the reference allele has been observed). Write the output to a VCF file called \path !A-J.vcf!
\item Looking at \path !A-J.vcf!, what is the reference and SNP base at position 10005912?
\item What is the total read depth at 10005912?
\item What is the total number of reads supporting the SNP call at position 10005912?
\item What sort of event has been called at position 10002643?
\end{itemize}
\end{frame}


\subsection{Annotation}
\begin{frame}{Exercises: Calling}
\framesubtitle{Annotation}
\begin{itemize}
\item `\texttt{vcf-annotate}' (from vcftools) takes a VCF as input and produces an annotated VCF with soft filters applied (nothing is removed, just marked according to whether the filter passed or failed)
\item Run `\texttt{vcf-annotate}' on \path !A-J.vcf! to produce \path !A-J.annotated.vcf!, applying all filters with default values (hint: run `\texttt{vcf-annotate ----help}' to get usage information)
\item View the annotated VCF file (e.g. with `\texttt{less -S}')
\item Did the SNP at position 10006276 pass the filters? If not, why?
\item Why does the SNP at position 10151443 fail the filters?
\end{itemize}
\end{frame}

\subsection{Indexing and merging}
\begin{frame}[fragile]{Exercises: Calling}
\framesubtitle{Indexing and merging}
Much like BAM files, VCF files need to be indexed in order to be accessed efficiently. Our VCFs are already sorted, so we can use \texttt{bgzip} and \texttt{tabix} to block compress and index them.
\begin{itemize}
\item Block compress A-J.vcf using bgzip, yielding a file called `\path !A-J.vcf.gz!
\item Index the compressed VCF using tabix in `vcf' preset mode
\item What is the index file called?
\item Now block compress and index the file \path !NZO.19:10000000-11000000.vcf!
\item Finally, use \texttt{vcf-merge} to merge \path !A-J.vcf.gz! and \path !NZO.19:10000000-11000000.vcf.gz! into a file called \path !merged.vcf!
\item View the merged VCF file (e.g. using \texttt{zless})
\item How many samples are there in the merged VCF file? (hint: check the number of sample columns)
\end{itemize}
\end{frame}





%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Summary
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Summary}
% n.b. use \section* to keep out of TOC

\begin{frame}{Summary}

  % Keep the summary *very short*.
  \begin{itemize}
  \item
    Basic workflow: alignment, BAM improvement, QC, variant calling
  \item
    Software used:
    \begin{itemize}
    \item Burrows-Wheeler Aligner (bwa; \url{http://bio-bwa.sourceforge.net/}) % -- newer sources are on github: \url{https://github.com/lh3/bwa})
    \item Picard (\url{http://picard.sourceforge.net/})
    \item samtools, bcftools, bgzip, tabix (\url{http://htslib.org/})
    \item vcftools (\url{http://vcftools.sourceforge.net/})
    \item IGV (\url{http://broadinstitute.org/igv/})
    \end{itemize}
%  \item
%    The \alert{second main message} of your talk in one or two lines.
%  \item
%    Perhaps a \alert{third message}, but not more than that.
  \end{itemize}

  % The following outlook is optional.
%  \vskip0pt plus.5fill
%  \begin{itemize}
%  \item
%    Outlook
%    \begin{itemize}
%    \item
%      Something you haven't solved.
%    \item
%      Something else you haven't solved.
%    \end{itemize}
%  \end{itemize}
\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Installation
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\section*{Appendix: Software Installation}
%
%
%%\subsection{curl}
%%\begin{frame}[fragile]{Installation}
%%\framesubtitle{curl}
%%A command-line utility and library for downloading files via http(s).
%%\begin{tcblisting}{title={Install curl}, listing only, listing options={style=custombash}}
%%# download the source from the curl website
%%cd ~
%%wget http://git-core.googlecode.com/files/git-1.8.2.tar.gz
%%# unpack
%%tar zxvf git-1.8.2.tar.gz
%%# build
%%cd git-1.8.2
%%./configure --prefix=$(echo ~/local)
%%make
%%# install
%%make install
%%\end{tcblisting}
%%
%%Prerequisites:
%%% sudo apt-get install gcc
%%gcc
%%% sudo apt-get install make
%%make
%%\end{frame}
%
%
%%\subsection{git}
%%\begin{frame}[fragile]{Installation}
%%\framesubtitle{git}
%%\begin{tcblisting}{title={Install git}, listing only, listing options={style=custombash}}
%%# download the source from google code
%%cd ~
%%wget http://git-core.googlecode.com/files/git-1.8.2.tar.gz
%%# unpack
%%tar zxvf git-1.8.2.tar.gz
%%# build
%%cd git-1.8.2
%%./configure --prefix=$(echo ~/local)
%%make
%%# install
%%make install
%%\end{tcblisting}
%%
%%Prerequisites:
%%% sudo apt-get install gcc
%%gcc
%%% sudo apt-get install make
%%make
%%\end{frame}
%
%
%
%\subsection*{Environment and Prerequisites}
%\begin{frame}[fragile]{Appendix: Software Installation}
%\framesubtitle{Setup Environment}
%\begin{tcblisting}{title={Setup Environment}, listing only, listing options={style=custombash}}
%####################################################
%# set environment variables
%####################################################
%
%# add ~/local/bin to PATH
%export PATH=~/local/bin:$PATH
%
%# add ~/local/lib/perl5 to PERL5LIB
%export PERL5LIB=~/local/lib/perl5:$PERL5LIB
%
%\end{tcblisting}
%\end{frame}
%
%
%\begin{frame}[fragile]{Appendix: Software Installation}
%\framesubtitle{Install Prerequisitites}
%\begin{tcblisting}{title={Install Prerequisites}, listing only, listing options={style=custombash}}
%####################################################
%# install prerequisite packages
%# (names may be specific to Ubuntu Linux >= 12.04)
%####################################################
%
%sudo apt-get update
%# these core utilities are probably already installed
%sudo apt-get install bash coreutils make g++ gcc perl wget
%# java is also probably already installed
%sudo apt-get install ant default-jdk default-jre icedtea-netx
%# these SCM tools may already be installed
%sudo apt-get install git subversion
%# development headers for samtools and bwa
%sudo apt-get install ncurses-dev zlib1g-dev
%# gnuplot for plotting in plot-bamcheck
%sudo apt-get install gnuplot
%
%####################################################
%# install useful utilities
%####################################################
%
%# less for viewing text
%sudo apt-get install less
%# mirage for viewing images
%sudo apt-get install mirage
%\end{tcblisting}
%\end{frame}
%
%
%\subsection*{bwa}
%\begin{frame}[fragile]{Appendix: Software Installation}
%\framesubtitle{bwa}
%\begin{tcblisting}{title={Install Development Version of bwa}, listing only, listing options={style=custombash}}
%# download the source from github
%cd ~
%git clone https://github.com/lh3/bwa.git
%
%# build
%cd bwa
%make
%
%# install
%install -s -D ./bwa ~/local/bin/bwa
%\end{tcblisting}
%
%Prerequisites:
%% sudo apt-get install gcc
%gcc,
%% sudo apt-get install make
%make,
%% sudo apt-get install zlib1g-dev
%zlib
%
%% may need CFLAGS=-msse2 on some machines
%\end{frame}
%
%
%\subsection*{samtools}
%\begin{frame}[fragile]{Appendix: Software Installation}
%\framesubtitle{samtools}
%\begin{tcblisting}{title={Install Development Version of samtools}, listing only, listing options={style=custombash}}
%# download the source from github
%cd ~
%git clone https://github.com/samtools/samtools.git
%
%# build
%cd samtools
%make
%
%# install
%install -s -D ./samtools ~/local/bin/samtools
%install -s -D ./bcftools/bcftools ~/local/bin/bcftools
%install -s -D ./misc/bamcheck ~/local/bin/bamcheck
%install -D ./misc/plot-bamcheck ~/local/bin/plot-bamcheck
%\end{tcblisting}
%
%Prerequisites:
%% sudo apt-get install gcc
%gcc,
%% sudo apt-get install make
%make,
%% sudo apt-get install zlib1g-dev
%zlib,
%% sudo apt-get install ncurses-dev
%ncurses,
%% sudo apt-get install perl
%perl, % (for plot-bamcheck),
%% sudo apt-get install gnuplot
%gnuplot %(for plot-bamcheck)
%
%\end{frame}
%
%
%
%\subsection*{bgzip/tabix}
%\begin{frame}[fragile]{Appendix: Software Installation}
%\framesubtitle{bgzip/tabix}
%\begin{tcblisting}{title={Install Development Version of bgzip/tabix}, listing only, listing options={style=custombash}}
%# download the source from github
%cd ~
%git clone https://github.com/samtools/tabix.git
%
%# build
%cd tabix
%make
%
%# install
%install -s -D ./bgzip ~/local/bin/bgzip
%install -s -D ./tabix ~/local/bin/tabix
%\end{tcblisting}
%
%Prerequisites:
%% sudo apt-get install gcc
%gcc,
%% sudo apt-get install make
%make,
%% sudo apt-get install zlib1g-dev
%zlib,
%% sudo apt-get install ncurses-dev
%ncurses,
%% sudo apt-get install perl
%perl, % (for plot-bamcheck),
%% sudo apt-get install gnuplot
%gnuplot %(for plot-bamcheck)
%
%\end{frame}
%
%
%\subsection*{vcftools}
%\begin{frame}[fragile]{Appendix: Software Installation}
%\framesubtitle{vcftools}
%%# download latest version of vcftools
%%wget -O vcftools_latest.tar.gz http://sourceforge.net/projects/vcftools/files/latest/download?source=files
%\begin{tcblisting}{title={Install Development Version of vcftools}, listing only, listing options={style=custombash}}
%# download the source from sourceforge
%cd ~
%svn checkout http://svn.code.sf.net/p/vcftools/code vcftools
%
%# build
%cd vcftools
%make
%
%# install scripts
%for file in $(ls ./bin); \
%  do install -D ./bin/${file} ~/local/bin/${file}; \
%done
%# install libraries
%for file in $(ls ./lib/perl5/site_perl); \
%  do install -D ./lib/perl5/site_perl/${file} ~/local/lib/perl5/${file}; \
%done
%# install vcftools binary
%install -s -D ./bin/vcftools ~/local/bin/vcftools
%\end{tcblisting}
%
%Prerequisites:
%% sudo apt-get install g++
%g++
%% sudo apt-get install make
%make
%
%\end{frame}
%
%
%\subsection*{Picard}
%\begin{frame}[fragile]{Appendix: Software Installation}
%\framesubtitle{Picard}
%\begin{tcblisting}{title={Install Development Version of Picard}, listing only, listing options={style=custombash}}
%# download the source from sourceforge
%cd ~
%svn checkout http://svn.code.sf.net/p/picard/code/trunk picard
%
%# build
%cd picard
%ant -lib lib/ant package-commands
%
%# install
%mkdir -p ~/local/bin/
%install -D ./dist/*.jar ~/local/bin/
%\end{tcblisting}
%
%Prerequisites:
%% sudo apt-get install default-jdk
%Java Development Kit (JDK), \\
%% sudo apt-get install java
%Java Runtime Environment (JRE),
%% sudo apt-get install ant
%ant
%
%\end{frame}
%
%
%\subsection*{IGV}
%\begin{frame}[fragile]{Appendix: Software Installation}
%\framesubtitle{Install \& Launch IGV}
%\begin{tcblisting}{title={Install \& Launch IGV}, listing only, listing options={style=custombash}}
%# launch IGV using Java Web Start
%# (also opens GUI and offers to install shortcut on Desktop)
%javaws http://www.broadinstitute.org/igv/projects/current/igv.jnlp
%\end{tcblisting}
%\begin{itemize}
%\item Select `Always trust content from this publisher'
%\item Press the `Run' button
%\item Press the `Allow' button to allow IGV to install a shortcut on the Desktop
%\end{itemize}
%Prerequisites:
%% sudo apt-get install java
%Java Runtime Environment (JRE), \\
%% sudo apt-get install icedtea-netx
%JNLP web start (javaws)
%\end{frame}






\end{document}
